src/lib/core/datetime.h

```diff
@@ -70,6 +71,33 @@ struct datetime {
 	int16_t tzindex;
 };
 
+/**
+ * To be able to perform arithmetics on time intervals and receive
+ * deterministic results, we have to keep months and years separately
+ * from seconds.
+ * Weeks, days, hours and minutes, all could be _precisely_ converted
+ * to seconds, but it's not the case for months (which might be 28, 29,
+ * 30, or 31 days long), or years (which could be leap year or not).
+ * Approach used here - to add/subtract months or years intervals only
+ * at the moment when we have particular date we operate on.
+ * Determinism of results is achieved due to the order we apply
+ * operations (from larger to smaller quantities):
+ * - years, then months, then weeks, days, hours, minutes,
+ *   seconds, and nanoseconds.
+ */
+struct datetime_interval {
```

Comment:
In general "interval" might have not only date/time connotation, but be related to distance/space. We do not have namespaces in C like in C++, so consider this prefix as sort of namespaces name.

But after discussion with others I've concluded that "interval" will be usually considered as time-related and this will be distinct enough.

So I've renamed it.